components/config-mgmt-service/backend/elastic/suggestions_internal_test.go

```diff
@@ -0,0 +1,42 @@
+package elastic
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateCaseInsensitivePatternEmpty(t *testing.T) {
+	regex := createCaseInsensitivePattern("")
+	assert.Equal(t, ".*.*", regex)
+}
+
+func TestCreateCaseInsensitivePatternWord(t *testing.T) {
+	regex := createCaseInsensitivePattern("bob")
+	assert.Equal(t, ".*[bB][oO][bB].*", regex)
+}
+
+func TestCreateCaseInsensitivePatternWordWithPunctuation(t *testing.T) {
```

Comment:
are there any characters this should fail on? maybe add some unhappy input tests here